core/src/main/scala/spinal/core/internals/ComponentEmitterVerilog.scala

```diff
@@ -1728,8 +1736,36 @@ end
     }
   }
 
+  var outputSignalNoUse: Set[BaseType] = Set()
+  if(!spinalConfig.emitFullComponentBindings) {
+    component.children.foreach(sub =>
+      sub.getAllIo
+      .foreach(io => if(io.isOutput) {
+        outputSignalNoUse = outputSignalNoUse + io
+      }
+    ))
+  }
+  component.dslBody.walkStatements {
```

Comment:
Shoulden't it be in the above if statement ? to avoid doing it when it isn't necessary.